урок JSX+CSS

Evgeny2015 opened the pull request:
* Компонент модального окна - "shared/modal-window"
* Компонент Layout - "shared/layout"
* Компонент Header - "shared/header"
* Компонент Logo - "shared/logo"
* Компонент краткого отображения операции - "shared/short-operation"
* Компонент полного отображения операции - "shared/edit-operation"
* Кнопка "в корзину" - "shared/add-to-basket-button"
* Компонент краткого отображения товара - "shared/short-product"
* Компонент полного отображения товара - "shared/long-product"
* Компонент товара для корзины - "shared/basket-product"

Evgeny2015 committed:
feat: append about myself in app.tsc

Evgeny2015 committed:
fix: change text-align in app.css

Evgeny2015 committed:
Updates

Evgeny2015 committed:
fix: append fio to app.tsx

Evgeny2015 committed:
Merge branch 'gh-pages' of github.com:Evgeny2015/evgeny2015.github.io into gh-pages

Evgeny2015 committed:
Updates

Evgeny2015 committed:
feat: create random Product and Operation

Evgeny2015 committed:
fix: use productList.length in createRandomProduct

Evgeny2015 committed:
fix: correct function types

Evgeny2015 committed:
Update main.yml

Evgeny2015 committed:
feat: new jsx+css components

AlexandrLitkevich commented on src/shared/add-to-basket-button/add-to-basket-button.tsx:
```diff
@@ -0,0 +1,46 @@
+ * если счетчик равен нулю, это кнопка, если больше нуля,
+ * то это инпут с отображением количества товара и двумя кнопками по краям (увеличения/уменьшения).
+ */
+const AddToBasketButton: FC<AddToBasketButtonProps> = ({ count, ...props }) => {
+  return (
+    <div className={cn('button-container')}
+        {...props}
+    >
```
```suggestion
    <div className={cn('button-container')} >

```

AlexandrLitkevich replied:
```suggestion
const AddToBasketButton: FC<AddToBasketButtonProps> = ({ count }) => {
  return (
    <div className={cn('button-container')}>

```

AlexandrLitkevich commented on src/shared/add-to-basket-button/add-to-basket-button.tsx:
```diff
@@ -0,0 +1,46 @@
+                Добавить
+        </button>
+
+        <div className={cn('button-container', {'invisible': count == 0 })}>
```
```{'invisible': count == 0 }``` стоит вынести в отдельную переменную выше ,  читаемость лучше

AlexandrLitkevich replied:
Другой момент что не нужно скрывать стилями достаточно использовать
``` 
if (!count) {
   return  null
}

```

AlexandrLitkevich commented on src/shared/basket-product/basket-product.tsx:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Компонент отображения товара для корзины
+ */
+const ShortProduct: FC<BasketProductProps> = ({ price, image, name, ...props }) => {
```
```...props``` конструкция тут не требуется, это используется не много в других ситуациях в лекции Паттерны вам будут рассказывать

AlexandrLitkevich commented on src/shared/header/header.tsx:
```diff
@@ -0,0 +1,26 @@
+/**
+ * Header component
+ */
+const Header: FC<HeaderProps> = ({ ...props }) => {
```
...props убрать стоит во всем проекте

Evgeny2015 committed:
feat: add theme and language support

Evgeny2015 committed:
Update src/shared/add-to-basket-button/add-to-basket-button.tsx

Co-authored-by: Alexsandr Litkevich <50555742+AlexandrLitkevich@users.noreply.github.com>

Evgeny2015 committed:
fix: removed unused props and styles

Evgeny2015 committed:
Merge pull request #2 from Evgeny2015/gh-pages

merge

Evgeny2015 committed:
Merge pull request #3 from Evgeny2015/homework1

Homework1

Evgeny2015 committed:
Merge branch 'main' into hooks

Evgeny2015 committed:
feat: add IntersectionObject, createPortal

Evgeny2015 committed:
fix: move language and theme providers to its own components

Evgeny2015 committed:
fix: move modal window to body

Evgeny2015 committed:
dev: add range-slider

Evgeny2015 committed:
Merge pull request #6 from Evgeny2015/hooks-hotfix

fix: move language and theme providers to its own components

Evgeny2015 committed:
fix: add open and close buttons

Evgeny2015 committed:
add forms

Evgeny2015 committed:
fix: range slider

Evgeny2015 committed:
Merge remote-tracking branch 'origin/hotfix' into forms

Evgeny2015 committed:
Merge remote-tracking branch 'origin/hooks' into forms

Evgeny2015 committed:
Merge remote-tracking branch 'origin/hooks-hotfix' into forms

Evgeny2015 committed:
Merge branch 'slider-hotfix' into forms

Evgeny2015 committed:
fix: modal-window

Evgeny2015 committed:
Merge pull request #4 from Evgeny2015/hooks

feat: add theme and language support

Evgeny2015 committed:
Merge pull request #5 from Evgeny2015/portal

feat: add IntersectionObject, createPortal

Evgeny2015 committed:
Merge pull request #7 from Evgeny2015/portal-hotfix

fix: move modal window to body

Evgeny2015 committed:
Merge pull request #8 from Evgeny2015/slider

dev: add range-slider

Evgeny2015 committed:
Merge pull request #10 from Evgeny2015/forms

add forms

Evgeny2015 committed:
Merge remote-tracking branch 'origin/hooks'

Evgeny2015 committed:
Merge remote-tracking branch 'origin/portal'

Evgeny2015 committed:
Merge remote-tracking branch 'origin/portal-hotfix'

Evgeny2015 committed:
Merge remote-tracking branch 'origin/slider'

Evgeny2015 committed:
Merge branch 'forms'

Evgeny2015 committed:
fix: slider min/max

Evgeny2015 committed:
dev: add AccountService and Jest test

Evgeny2015 committed:
fix: errors

Evgeny2015 committed:
dev: add pages

Evgeny2015 committed:
dev: add routes and redis with thunk

Evgeny2015 committed:
fix: move routes in component and add dialog route

Evgeny2015 committed:
Merge pull request #15 from Evgeny2015/redux

dev: add routes and redis with thunk

Evgeny2015 committed:
Merge pull request #16 from Evgeny2015/hotfix

fix: move routes in component and add dialog route

Evgeny2015 committed:
Revert "fix: move routes in component and add dialog route"

Evgeny2015 committed:
Merge pull request #17 from Evgeny2015/revert-16-hotfix

Revert "fix: move routes in component and add dialog route"

Evgeny2015 committed:
Merge remote-tracking branch 'origin/redux'

Evgeny2015 committed:
Merge remote-tracking branch 'origin/routes'

Evgeny2015 committed:
fix: move routes in component and add dialog route

# Conflicts:
#	src/app/App.tsx
#	src/features/forms/ProductForm/ProductForm.stories.tsx
#	src/features/forms/ProductForm/ProductForm.tsx
#	src/pages/product/ProductPage.tsx

Evgeny2015 committed:
fix

Evgeny2015 committed:
fix: add empty basket message

Evgeny2015 committed:
fix: set relational paths for storybook

Evgeny2015 committed:
dev: add ProtectedRoute and Logout

Evgeny2015 committed:
dev: register new user with React Query and Redux-toolkit-query

Evgeny2015 committed:
Merge branch 'develop'

Evgeny2015 committed:
обработка ошибок

Evgeny2015 committed:
Merge branch 'develop'

Evgeny2015 committed:
add https://github.com/React-js-OTUS/otus-rest-server models

Evgeny2015 committed:
dev: add orders page, modify products page

Evgeny2015 committed:
Merge branch 'develop'
